x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/ClientVersion.java

```diff
@@ -87,7 +88,11 @@ static Manifest getManifest(URL url) throws IOException {
             if (url.getProtocol().equals("jar")) {
                 JarURLConnection jarConn = (JarURLConnection) conn;
                 if (jarConn.getEntryName() == null) { // the URL points to a JAR file
-                    return jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    Manifest manifest = jarConn.getManifest(); // in case of a fat JAR, this would return the outermost JAR's manifest
+                    try {
+                        jarConn.getJarFile().close(); // prevent locked file errors in Windows. The Manifest has been read by now
```

Comment:
Here both `getJarFile()` and `.close()` can throw an IOException. `getManifest()` call above does use `getJarFile()` so the danger of an IOException being thrown while calling `getJarFile()` wouldn't be ignored. But, would make sense to call `jarConn.getJarFile()` and reuse this object in both `getManifest()` and `.close()` calls?

Reply:
> why is this .close() needed now, but wasn't needed before?

We just didn't test with JAR non-entry URLs. The tests were using file (i.e. setting the `ClientVersion.CURRENT`) and entry JAR URLs (i.e. the JarInJar test).

> Here both getJarFile() and .close() can throw an IOException. 

True. And while the former's exception should surface, the latter (if it can ever occur) is non-critical.

> would make sense to call jarConn.getJarFile() and reuse this object in both getManifest() and .close() calls?

That exposes a bit more of "deprecated" surface and is more verbose, but I see how this would make sense. I've applied your suggestion.